src/main/java/org/killbill/billing/plugin/stripe/dao/StripeDao.java

```diff
@@ -222,8 +224,9 @@ public StripeResponsesRecord withConnection(final Connection conn) throws SQLExc
                                                  asString(additionalDataMap),
                                                  toLocalDateTime(utcNow),
                                                  kbTenantId.toString())
-                                         .returning()
-                                         .fetchOne();
+                                         .execute();
+                               final long lastId = dslContext.lastID().longValue();
```

Comment:
The cast might not work on all engines. Can you just get a `Field` (`getValue `) and use jOOQ's `fetchInto`?

Reply:
@pierre I'm little bit confused with 

> .... get a Field (getValue ) and use jOOQ's fetchInto 

would you elaborate more? AFAIK, after `execute()`, we need last inserted ID to get correct `STRIPE_RESPONSES` data, thus it would looks like:

```java
final BigInteger lastID = DSL.using(config).lastID(); // change to BigInteger instead of long
return DSL.using(configuration).fetchOne(STRIPE_RESPONSES, STRIPE_RESPONSES.RECORD_ID.eq(ULong.valueOf(lastId)));
```

Reply:
Would something like that work? Basically don't assume the `record_id` is a specific type (e.g. `long`):

```
return DSL.using(conn, dialect, settings).transactionResult(configuration -> {
  final DSLContext dslContext = DSL.using(configuration);
  dslContext.insertInto(STRIPE_RESPONSES,
                       STRIPE_RESPONSES.KB_ACCOUNT_ID,
                       STRIPE_RESPONSES.KB_PAYMENT_ID,
                       STRIPE_RESPONSES.KB_PAYMENT_TRANSACTION_ID,
                       STRIPE_RESPONSES.TRANSACTION_TYPE,
                       STRIPE_RESPONSES.AMOUNT,
                       STRIPE_RESPONSES.CURRENCY,
                       STRIPE_RESPONSES.STRIPE_ID,
                       STRIPE_RESPONSES.ADDITIONAL_DATA,
                       STRIPE_RESPONSES.CREATED_DATE,
                       STRIPE_RESPONSES.KB_TENANT_ID)
           .values(kbAccountId.toString(),
                   kbPaymentId.toString(),
                   kbPaymentTransactionId.toString(),
                   transactionType.toString(),
                   amount,
                   currency == null ? null : currency.name(),
                   stripePaymentIntent == null ? null : stripePaymentIntent.getId(),
                   asString(additionalDataMap),
                   toLocalDateTime(utcNow),
                   kbTenantId.toString())
           .execute();

  return dslContext.selectFrom(STRIPE_RESPONSES)
                   .where(STRIPE_RESPONSES.RECORD_ID.equal(STRIPE_RESPONSES.RECORD_ID.getDataType().convert(dslContext.lastID())))
                   .fetch();
}
```